Tabby/components/camera/CameraModel.tsx

```diff
@@ -49,23 +64,117 @@ const CameraModal: React.FC<CameraModalProps> = ({ closeModal }) => {
             mediaTypes: ['images'],
             allowsEditing: true,
             quality: 1,
-            base64: true,
         });
 
-        setIsProcessing(false);
         if (!result.canceled) {
-            const resultBase64 = result.assets[0].base64; // base64 string to send to backend
-            console.log(resultBase64);
-            closeModal();
+            const returnedBooks = await uploadImage(result.assets[0].uri);
+            if (returnedBooks)
+                await userPickBook(returnedBooks);
+        }
+        setIsProcessing(false);
+    };
+
+    // for testing activity indicator
+    async function sleep(ms: number): Promise<void> {
+        return new Promise((resolve) => setTimeout(resolve, ms));
+    }
+
+    // Opens modal for user to select correct book
+    const userPickBook = async (bookArr: Book[]) => {
+        await setUserChoosing(true);
+
+        // DONT REMOVE THIS SLEEP
+        // idk why but if you remove it then shit breaks
+        await sleep(1000);
+
+        if (onBookSelectionStart) {
+            onBookSelectionStart(bookArr);
+        }
+    }
+
+    // uploads image to scan_cover endpoint
+    const uploadImage = async (imageUri: string) => {
+        try {
+            console.log("Sending koyeb image");
+
+            // convert image to blob raw data
+            const res = await fetch(imageUri);
+            const blob = await res.blob();
+
+            // fetch scan_cover
+            const response = await fetch('https://just-ulrike-tabby-app-9d270e1b.koyeb.app/books/scan_cover', {
```

Comment:
Caesar mentioned about putting the domain into an env file so it can be easily changed. We can probably do that here too and just use the default value if non is given to this domain